### PR TITLE
[#12222] Instructor help page: Page not scrolling to the right section

### LINKED ISSUE #12222: The page menu is not working as intended
**Steps to reproduce**

You need to log in as an instructor and navigate to the help page.

**Expected behaviour**

When i click on one of the links, it should scroll to the right section.

**Actual behaviour**

It is scrolling to the top by some pixels (weird behavior).

image

**Additional info**

### PR DESCRIPTION
**Pull request by noreddinelam**
Fixes #12222

**Outline of Solution**

To resolve this issue, I use to calculate the y coordinate of the section and I used the scrollTo function directly to do it.
The problem was in these two lines :
`el.scrollIntoView();
 window.scrollBy(0, -50);`
 
 The second instruction is not waiting for the first one to finish and do the scroll by -50px.
 
 I removed the '#helpPage' because it is not necessary to get all the children and then search for the section using the id.

### CODE REVIEW

**Comment by noreddinelam** [comment]
Still waiting for review

**Comment by samuelfangjw** [comment]
> Still waiting for review

Open source maintainers on this project are unpaid volunteers with their own work commitments as well. Please be patient on this.

**Review by weiquu — COMMENTED** [review]
Hi @noreddinelam, sorry for the delay! The fix works, but I'm just wondering if it would be better to use a more standard Bootstrap approach. You can refer to [this](https://mdbootstrap.com/docs/standard/extended/back-to-top/) (in particular the .js code)

Edit: Sorry please ignore and refer to @zhaojj2209's reply (:

**Comment by zhaojj2209** [comment]
TEAMMATES makes use of `PageScrollService` from the [ngx-page-scroll-core](https://github.com/Nolanus/ngx-page-scroll) library, not sure why this wasn't used here previously. Maybe using this would help to simply things?

**Comment by noreddinelam** [comment]
@zhaojj2209 Yeah I noticed that. I tried to use it but unfortunately, It was not working and it was scrolling by like 1cm.

As mentioned here ; (an issue opened in the official project) : 
image

The issue [link ](https://github.com/Nolanus/ngx-page-scroll/issues/434) if u want to read all the discussion.

Now, I found a solution and what was the problem. As mentioned in the issue, they said that the scrolling behavior `smooth `is not working with ngx-page-scroll. The default behavior when we use bootstrap, is smooth and as consequence, all the scrolling mechanism when u touch a link was not working in the application.

Here is the solution : 
Set the scroll behavior to auto in the style.scss file like this :

`html {
  scroll-behavior: auto !important;
}`

And then we can use the ngx-page-scroll.

If u agree with the solution, I can provide a new version of the code.

Feel free to give me more hints.

**Review by zhaojj2209 — APPROVED** [review]
LGTM, thanks for the changes!

**Comment by noreddinelam** [comment]
Thanks, for the approval.

Just a question i think that my branch is out-of-date with the master branch so I don't know If I should click on update branch let you do it when you merge.

File: src/web/styles.scss
Comment (noreddinelam) [reply src/web/styles.scss:4]: yeah of course.

Comment (noreddinelam) [reply src/web/styles.scss:4]: @samuelfangjw as i mentioned earlier in the discussion, we need to use scroll behavior auto so that ngx-scroll-page would work. I will add an explanation on the code and push it.

Comment (noreddinelam) [reply src/web/styles.scss]: @samuelfangjw it is done.

**Pull request merged**
